gateway-ha/src/main/java/io/trino/gateway/ha/router/RuleReloadingRoutingGroupSelector.java

```diff
@@ -43,10 +44,12 @@ public class RuleReloadingRoutingGroupSelector
     private volatile Rules rules = new Rules();
     private volatile long lastUpdatedTime;
     private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock(true);
+    private final ProcessedRequestConfig processedRequestConfig;
```

Comment:
I was wondering what's the best way to make this flow optional. At least till it gets some mileage on it. Overall it will be a good idea to skip the processing of the request if we want have a more performant flow.